src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = null;
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            machines = new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            machines = new TruckProducer().get();
+        }
+
+        if (machines == null) {
+            return new ArrayList<>();
```

Comment:
you may initialize machines with new ArrayList and remove this check
or make return from each if statement and if none worked return new ArrayList (again without additional check)